src/TractionRec.php

```diff
@@ -109,7 +109,9 @@ public function loadProgramCategoryTags(): array {
         TREX1__Program_Category_Tag__c.TREX1__Program_Category__r.id,
         TREX1__Program_Category_Tag__c.TREX1__Program_Category__r.name,
         TREX1__Program_Category_Tag__c.TREX1__Program_Category__r.TREX1__Available__c
-        FROM TREX1__Program_Category_Tag__c WHERE TREX1__Program__r.TREX1__Available_Online__c = true AND TREX1__Program_Category__r.TREX1__Available_Online__c = true'
+        FROM TREX1__Program_Category_Tag__c
```

Comment:
queries must be part of the config, not hardcoded

Reply:
@podarok could you please describe more? This is a query to Salesforce to get data for API.
It gets all available products that are ready for PEF. It uses SF's SQOOL to get data. What variables do you want to put to the config?

Reply:
@anpolimus instead of hardcoding quesries into PHP code we need to put them into Drupal config, in order to be able to change it on the fly without additional deployment of the code @anpolimus @froboy

Reply:
@podarok I feel like putting the entire query into config could be a recipe for disaster. It's likely that if the query is changing that significant underlying migration config will also need to change. We've confirmed that this is the "most standard" implementation from Traction Rec, and since this is all OOP, I think custom implementations could do something like `class CustomTractionRec extends TractionRec` and then write their own version of `TractionRec::loadProgramCategoryTags`, right?

Reply:
Ok. Let's see how it goes with the current code ;)